### DISCUSSION TITLE
Changing Databus (DB) pinout in source code

### DISCUSSION THREAD

**Discussion by tiff-lee** [erichelgeson/BlueSCSI-v1#272] [Q&A]
Does anyone have any idea how to change the databus pinout in the bluescsi source?
I want to change the DB pinout on the STM32 to align better with something i'm trying and in the source file it seems all those pins are commented out?

I have no experience with STM32 stuff only arduino, I would of thought they would be uncommented and it would be a case of just changing the pin allocation but I also can't find any other reference to those pins in the source. I'm clearly missing something because obviously if you download the pre-compiled bin it works.

Cheers

<img width="307" alt="1718959334059" src="https://github.com/user-attachments/assets/4b52280c-748f-4aa0-b2a9-b6f1cc9bce54">

**Comment by erichelgeson** [comment]
We directly write to the "BSRR" (bit set-reset register) to write data values

There's a BSRR lookup table here: https://github.com/erichelgeson/BlueSCSI/blob/main/src/BlueSCSI.cpp#L358

To change the data pins you would need to update the BSRR lookup table.

**Reply by erichelgeson** [reply]
I might add too if you're looking to do new development on BlueSCSI you might want to checkout v2 at this point, unless there is some specific use case (which I'd love to know about)

**Reply by tiff-lee** [reply]
Hi Eric, thanks for the reply and apologies for raising this as an 'issue' and not a discussion (I really need to get my head around github more).

So as far as specific use case goes it is for the Capcom CPS3 arcade board which the 1.1a in its stock configuration works absolutely fine for.  I wanted to change around some of the databus pin assignments as I was removing the 4 Bourns resistors networks and swapping them for the bourns single DIP package instead but I also intend to swap the 50 pin connector for a female header so it plugs directly into the back of the board. 

Essentially I wanted to retain the low cost of the stock setup but just compact it a little so it plugs directly into the rear of of the board and fits inside some of the steel cases available for the CPS3 itself.
